cmake/CPM.cmake

```diff
@@ -46,6 +46,13 @@ set(CURRENT_CPM_VERSION 1.0.0-development-version)
 
 get_filename_component(CPM_CURRENT_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
 if(CPM_DIRECTORY)
+  if(${CMAKE_VERSION} VERSION_LESS "3.17.0")
+    get_directory_property(hasParent PARENT_DIRECTORY)
+    if(NOT CPM_DIRECTORY STREQUAL CPM_CURRENT_DIRECTORY OR ${hasParent})
```

Comment:
Could you add a comment above this explaining what issue this check fixes? 

Also I feel some parenthesis could be useful around the `NOT` expression as I'm confused by CMake's operator precedence.